include/ClientImpl.h

```diff
@@ -160,14 +167,40 @@ class ClientImpl {
     void createPendingConnection(const std::string &hostname, const unsigned short port, const int64_t time=0);
     void erasePendingConnection(PendingConnection *);
 
+    /**
+     * Generates hash-digest for the for the password. Supported hash functions are
+     * SHA-1 and SHA-256
+     * @param: password for which hash digest will be generated
+     * @throws MDHashException: An error occurred generating hash for the password
+     */
+    void hashPassword(const std::string& password) throw (MDHashException);
+
+    /**
+     * Initializes SSL library, contexts and algorithms to use
+     * @throws SSLException: An error occurred during initialization of SSL
+     */
+    void initSsl() throw (SSLException);
+    inline void notifySslClose(bufferevent *bev) {
+        SSL *ssl = bufferevent_openssl_get_ssl(bev);
+        if (ssl == NULL) {
+            return;
+        }
+        if (SSL_get_shutdown(ssl) & SSL_RECEIVED_SHUTDOWN) {
```

Comment:
True. SSL shutdown is has only one bit set, so above will also work but safer to as u suggested. will update it